contracts/vending-minter/src/validation.rs

```diff
@@ -0,0 +1,41 @@
+use crate::ContractError;
+use crate::ContractError::CheckedMultiplyFractionError;
+use cosmwasm_std::Uint128;
+
+pub fn get_three_percent_of_tokens(num_tokens: u32) -> Result<Uint128, ContractError> {
+    let three_percent = (Uint128::new(3), Uint128::new(100));
+    let three_percent_tokens = Uint128::from(num_tokens)
+        .checked_mul_ceil(three_percent)
+        .map_err(|_| CheckedMultiplyFractionError {})?;
```

Comment:
There's no way to generate an error here actually, because the fraction is locked to 3/100. The .map_err is there because Rust requires an error catch. But would only get an error if You did something like (Uint128::new(3), Uint128::new(0)); (division by 0 error).